pkg/controller/certificaterequests/sync.go

```diff
@@ -63,6 +63,12 @@ func (c *Controller) Sync(ctx context.Context, cr *cmapi.CertificateRequest) (er
 		return nil
```

Comment:
Should we not first check the RequestReadyReason before we try to call `c.reporter.Denied(crCopy)`?
Basically moving this section  https://github.com/cert-manager/cert-manager/pull/5887/files#diff-15766c9d9c1553a3397c9b8ec3506f9d19bcc5fde2a4f3238fa2d74d06dffcdaR79-R82 before the apiutil.CertificateRequestIsDenied(cr) check.

Reply:
I think the order does not really matter in most cases
I think we originally set it like this to ensure that the Denied reason always gets bubbled up to the CR as it should be what users see first and try to fix if a CR gets denied. There is a test case that explicitly tests that a failed reason gets overwritten with a denied reason https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/certificaterequests/sync_test.go#L222-L268. I don't really see a reason to swap it around

Reply:
Sorry for the confusion, I actually meant that we should check for `cmapi.CertificateRequestReasonDenied` before calling `c.reporter.Denied(crCopy)`. However, I just noted that that function does not trigger a Kubernetes Event, it only updates the Status. This means that it doesn't matter whether we check the condition or not.